cloud/pkg/dynamiccontroller/application/eventhandler.go

```diff
@@ -59,23 +62,27 @@ func NewHandlerCenter(informerFactory dynamicinformer.DynamicSharedInformerFacto
 }
 
 func (c *handlerCenter) ForResource(gvr schema.GroupVersionResource) *CommonResourceEventHandler {
-	var handler *CommonResourceEventHandler
-	if store, ok := c.handlers[gvr]; ok {
-		handler = store
-	} else {
-		klog.Infof("[metaserver/HandlerCenter] prepare a new resourceEventHandler(%v)", gvr)
-		handler = NewCommonResourceEventHandler(gvr, c.dynamicSharedInformerFactory, c.messageLayer)
-		c.handlers[gvr] = handler
+	if handler, ok := c.handlers[gvr]; ok {
```

Comment:
just confusing about where to add lock. here not add a lock, but `ForResource` interface is also used by other modules

Reply:
good catch, updated